src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -603,11 +620,18 @@ public void updateSyntheticClassForSuperCall(Expression expr) {
           expr.asMethodCallExpr(),
           this.superClass,
           methodAndReturnType.getOrDefault(expr.asMethodCallExpr().getNameAsString(), ""));
-    } else {
+    } else if (expr instanceof FieldAccessExpr) {
       updateUnsolvedClassWithFields(
           expr.asFieldAccessExpr().getNameAsString(),
           superClass,
           classAndPackageMap.getOrDefault(superClass, this.currentPackage));
+    } else if (expr instanceof NameExpr) {
+      updateUnsolvedClassWithFields(
+          expr.asNameExpr().getNameAsString(),
+          superClass,
+          classAndPackageMap.getOrDefault(superClass, this.currentPackage));
```

Comment:
Since this block is the same as the one above, you should just check `expr instanceof FieldAccessExpr || expr instanceof NameExpr` and avoid duplicating this code.

Reply:
Professor, they are actually different. For `FieldAccessExpr`, the first parameter for the method `updateUnsolvedClassWithFields` is `expr.asFieldAcessExpr().getNameAsString()`, while for `NameExpr`, the first parameter is `expr.asNameExpr().getNameAsString()`

Reply:
Ah, you're right. Even so, there is too much duplicate code here, so it's hard for a reader to see what's different. I suggest adding one line to each block that computes the thing that's different (e.g., `String nameAsString = expr.asNameExpr().getNameAsString()` vs `expr.asFieldAcessExpr().getNameAsString()`) to draw the attention of a reader to the difference.